test/plugin_functional/test_suites/saved_objects_management/hidden_from_http_apis.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { join } from 'path';
+import expect from '@kbn/expect';
+import type { Response } from 'supertest';
+import { SavedObject } from '@kbn/core/types';
+import type { PluginFunctionalProviderContext } from '../../services';
+
+function parseNdJson(input: string): Array<SavedObject<any>> {
+  return input.split('\n').map((str) => JSON.parse(str));
+}
+
+export default function ({ getService }: PluginFunctionalProviderContext) {
+  const supertest = getService('supertest');
+  const kbnServer = getService('kibanaServer');
+  const esArchiver = getService('esArchiver');
+
+  describe('types with `hiddenFromHttpApis` ', () => {
+    before(async () => {
+      // `kbnServer.savedObjects.cleanStandardList` uses global saved objects `delete`.
```

Comment:
Not strictly needed but I explain why I'm using a combination of `kbnServer` and `esArchiver` for setup/teardown. Ideally, once https://github.com/elastic/kibana/issues/148412 is resolved, we can switch over completely to `kbnServer`.
An easy check is to uncomment this line and switch over from `esArchiver` to `kbnServer.importExport.unload` in the `after` hook.